drape/gl_extensions_list.cpp

```diff
@@ -37,6 +37,24 @@ void GLExtensionsList::Init(dp::ApiVersion apiVersion)
     SetExtension(VertexArrayObject, true);
     SetExtension(UintIndices, true);
   }
+#elif defined(OMIM_OS_LINUX)
+  SetExtension(MapBuffer, true);
+  SetExtension(UintIndices, true);
+  if (apiVersion == dp::ApiVersion::OpenGLES2)
+  {
+#if defined(OMIM_OS_LINUX_HAS_EGL)
+    SetExtension(VertexArrayObject, true); // Otherwise failure with offset
+    SetExtension(MapBuffer, true);
+    SetExtension(MapBufferRange, true);
+#else
+    CHECK(false, ("This build was compiled without EGL, hence OpenGLES2 is not supported in this build"));
+#endif  // OMIM_OS_LINUX_HAS_EGL
```

Comment:
Can you please add similar comments to other `#endif` clauses here for easier readability? It looks complex now.

Reply:
I think I would remove the dependency on EGL, and then this would not be so complex anymore.

Reply:
> Or, alternatively, should we rewrite/remove OMIM_OS_MOBILE to dynamically initialize GLES if it's present, and fall back to GL if it's not? 

I am not entirely sure what you mean by dynamically initialize GLES in this context, but `OMIM_OS_MOBILE` should probably be removed from here. And maybe even completely. In the past it made sense to use `OMIM_OS_MOBILE` for switching between the mobile and the desktop wikipedia base url, but by now it is the same everywhere.
So I would consider removing `OMIM_OS_MOBILE` completely in a separate PR. What do you think?

Reply:
Now on Android and iOS GLES is initialized via OMIM_OS_MOBILE

Should we do it differently now? GLES may not be available on desktop platforms. Btw, we can remove it completely on iOS too, it should use Metal instead. But for Android it can be used as a fallback from Vulkan for <24 APIs and also to work-around some Vulkan issues.

Reply:
Regarding Wikipedia, why do you think it's better to remove different links? One less redirect means a better UX.

Reply:
> Now on Android and iOS GLES is initialized via OMIM_OS_MOBILE

My observation is that for android it is iniitialized via it's own `OMIM_OS_ANDROID` so for GLES purposes `OMIM_OS_MOBILE` should probalby be replaced with something like `OMIM_OS_IPHONE`.

> One less redirect means a better UX

In my experience, If one opens the mobile version on a desktop, then there is no redirect, and the [latest desktop experience](https://wikimediafoundation.org/news/2023/01/18/wikipedia-gets-a-fresh-new-look-first-desktop-update-in-a-decade-puts-usability-at-the-forefront/) is pretty close to the mobile one. 
In any case, it might make sense to keep `OMIM_OS_MOBILE` for the wiki links, while the GLES init should probably be refactored.

Reply:
Anyhow, now that macro directive has been removed.